CRM/Altinvoice/Hook/BuildForm/PreferencesContribute.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/**
+ * Class CRM_Altinvoice_Hook_BuildForm_PreferencesContribute
+ *
+ * Processes buildForm stage for the Civicontribute preferences form.
+ */
+class CRM_Altinvoice_Hook_BuildForm_PreferencesContribute {
+
+  /**
+   * @var \CRM_Admin_Form_Preferences_Contribute
+   */
+  private $form;
+
+  /**
+   * CRM_Altinvoice_Hook_BuildForm_PreferencesContribute constructor.
+   *
+   * @param \CRM_Admin_Form_Preferences_Contribute $form
+   */
+  public function __construct(CRM_Admin_Form_Preferences_Contribute $form) {
+    $this->form = $form;
+  }
+
+  /**
+   * Further processes buildform stage for the given form.
+   */
+  public function buildForm() {
+    $this->addFields();
+    $this->addTemplates();
+    $this->setDefaults();
+  }
+
+  /**
+   * Adds ne fields to the form.
+   */
+  private function addFields() {
+    $this->form->add('advcheckbox', 'include_link_to_pay', ts('Include Link to Pay Invoice When Emailing Invoices?'));
+  }
+
+  /**
+   * Adds templates to the form to show the new fields.
+   */
+  private function addTemplates() {
+    $templatePath = CRM_Altinvoice_ExtensionUtil::path() . '/templates';
```

Comment:
I don't think this class is defined?  Maybe there's a missing part to this PR?

Reply:
@MegaphoneJon, this is the class defined by civix when doing the scaffolding for the extension. You can find it here:

https://github.com/MegaphoneJon/com.megaphonetech.altinvoice/blob/c5428ef3efd0326877fa672a9bf4ab7c9162257c/altinvoice.civix.php#L9